pkg/provider/defaults/hardening-ubuntu-2204.yml

```diff
@@ -0,0 +1,115 @@
+---
+- name: Ubuntu 22.04 Hardening
+  hosts: all
+  become: true
+  gather_facts: yes
+
+  tasks:
+    - name: Apply hardening if OS is Ubuntu 22.04
+      block:
+        - name: Update and upgrade all packages to the latest version
+          ansible.builtin.apt:
+            upgrade: dist
+            update_cache: yes
+            cache_valid_time: 3600
+
+        # UFW
+        - name: Ensure UFW is installed
+          ansible.builtin.apt:
+            name: ufw
+            state: present
+        - name: Enable UFW
+          ansible.builtin.ufw:
+            state: enabled
+            policy: deny
+            direction: incoming
+        - name: Allow SSH in UFW
+          ansible.builtin.ufw:
+            rule: allow
+            name: OpenSSH
```

Comment:
How do we allow more? 
Create equivalent rules in additional playbooks for those ports?

Also, what happens if the host is not Ubuntu? Should this contain some conditional based on the `image` host variable?

There might be an example of this in the `upgrade.yml` I added to the private infra repo. Maybe it makes sense actually adding that entire playbook into the default providers here?